packages/astro/src/runtime/client/hmr.ts

```diff
@@ -1,38 +1,23 @@
 if (import.meta.hot) {
 	const parser = new DOMParser();
-	import.meta.hot.on('astro:reload', async ({ html }: { html: string }) => {
-		const { default: morphdom } = await import('morphdom');
-		const doc = parser.parseFromString(html, 'text/html');
+	import.meta.hot.on('astro:update', async ({ file }) => {
+		const { default: diff } = await import('micromorph');
```

Comment:
No docs yet but https://github.com/natemoo-re/micromorph is a diffing library I wrote that diffs real DOM nodes rather than VDOM trees. It's pretty small and fast, and works on full `document` nodes (since `head` elements need special consideration)